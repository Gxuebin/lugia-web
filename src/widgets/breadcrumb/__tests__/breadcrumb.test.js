import React from 'react';

import Breadcrumb from '../';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Icon from '../../icon';

Enzyme.configure({ adapter: new Adapter() });

const routes = [
  {
    path: 'index',
    breadcrumbName: '首页',
  },
  {
    path: 'first',
    breadcrumbName: '一级面包屑',
  },
  {
    path: 'second',
    breadcrumbName: '二级面包屑',
  },
  {
    path: 'third',
    breadcrumbName: '当前页面',
  },
];

describe('Breadcrumb', () => {
  it('<Breadcrumb>和<Breadcrumn.Item>配合使用', () => {
    expect(
      renderer
        .create(
          <Breadcrumb>
            <Breadcrumb.Item>主页</Breadcrumb.Item>
            <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 中separator分隔符', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item>主页</Breadcrumb.Item>
            <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 中传入图标', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item>
              <Icon iconClass="lugia-icon-reminder_close_circle_o" type="home" />
              <span>主页</span>
            </Breadcrumb.Item>
            <Breadcrumb.Item>
              <span>一级面包屑菜单</span>
              <Icon iconClass="lugia-icon-reminder_close_circle_o" type="home" />
            </Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb.Item> 中传入href属性，文字变成超链接', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item href="">主页</Breadcrumb.Item>
            <Breadcrumb.Item href="">一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item href="">二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 传入routes参数直接生成 面包屑', () => {
    expect(
      renderer.create(<Breadcrumb separator={'>'} routes={routes} />).toJSON()
    ).toMatchSnapshot();
  });

  it('最后一项也显示分隔符, lastSeparator', () => {
    expect(
      renderer.create(<Breadcrumb separator={'>'} lastSeparator={'>'} routes={routes} />).toJSON()
    ).toMatchSnapshot();
  });
});
