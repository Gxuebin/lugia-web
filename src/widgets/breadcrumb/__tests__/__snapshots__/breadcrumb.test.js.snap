// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb <Breadcrumb.Item> 中传入href属性，文字变成超链接 1`] = `
.c1 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

.c0 {
  color: #999;
  font-weight: 500;
  font-size: 1.1666666666666667em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #333;
}

.c2 {
  color: #333;
  font-weight: 500;
  font-size: 1.1666666666666667em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  color: #333;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <a
      className="sv_breadcrumb_link c0"
      href=""
    >
      主页
    </a>
    <span
      className="sv_breadcrumb_separator c1"
      href=""
    >
      &gt;
    </span>
  </span>
  <span>
    <a
      className="sv_breadcrumb_link c0"
      href=""
    >
      一级菜单
    </a>
    <span
      className="sv_breadcrumb_separator c1"
      href=""
    >
      &gt;
    </span>
  </span>
  <span>
    <a
      className="sv_breadcrumb_link c2"
      href=""
    >
      二级菜单
    </a>
    <span
      className="sv_breadcrumb_separator c1"
      href=""
    >
      
    </span>
  </span>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 中separator分隔符 1`] = `
.c0 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
}

.c1 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      主页
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      一级菜单
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      二级菜单
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      
    </span>
  </span>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 中传入图标 1`] = `
.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c0 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
}

.c2 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <i
        className="lugia-icon-reminder_close_circle_o  c1"
        onClick={undefined}
      />
      <span>
        主页
      </span>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <span>
        一级面包屑菜单
      </span>
      <i
        className="lugia-icon-reminder_close_circle_o  c1"
        onClick={undefined}
      />
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      
    </span>
  </span>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 传入routes参数直接生成 面包屑 1`] = `
.c0 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
}

.c2 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

.c1 {
  color: #999;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  color: #333;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index"
      >
        首页
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index/first"
      >
        一级面包屑
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index/first/second"
      >
        二级面包屑
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <span>
        当前页面
      </span>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      
    </span>
  </span>
</div>
`;

exports[`Breadcrumb <Breadcrumb>和<Breadcrumn.Item>配合使用 1`] = `
.c0 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
}

.c1 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      主页
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      /
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      一级菜单
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      /
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      二级菜单
    </span>
    <span
      className="sv_breadcrumb_separator c1"
    >
      
    </span>
  </span>
</div>
`;

exports[`Breadcrumb 最后一项也显示分隔符, lastSeparator 1`] = `
.c0 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
}

.c2 {
  font-weight: 500;
  font-size: 1.1666666666666667em;
  margin-left: 0.8333333333333334em;
  margin-right: 0.8333333333333334em;
  color: #999;
}

.c1 {
  color: #999;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  color: #333;
}

<div
  className="sv_breadcrumb_wrap"
>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index"
      >
        首页
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index/first"
      >
        一级面包屑
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <a
        className="c1"
        href="#/index/first/second"
      >
        二级面包屑
      </a>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
  <span>
    <span
      className="sv_breadcrumb_link c0"
    >
      <span>
        当前页面
      </span>
    </span>
    <span
      className="sv_breadcrumb_separator c2"
    >
      &gt;
    </span>
  </span>
</div>
`;
